### PR TITLE
*: Support user with any host(%)

### PR DESCRIPTION
**Pull request by shenli**
For tidb with docker, we can connect it with any host.

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -518,6 +518,21 @@ func (s *session) Close() error {
 	return s.FinishTxn(true)
 }
 
+func (s *session) getPassword(name, host string) (string, error) {
+	// Get password for name and host.
+	authSQL := fmt.Sprintf("SELECT Password FROM %s.%s WHERE User=\"%s\" and Host=\"%s\";", mysql.SystemDB, mysql.UserTable, name, host)
```
Comment (siddontang) [inline session.go]: using '%s' instead of \"%s\" is not ok?

**Comment by siddontang** [comment]
LGTM

File: session.go (hunk 1 above)
Comment (shenli) [inline session.go]: PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**